### PR TITLE
[5.1] Validation rule: required_unless

### PR DESCRIPTION
**Pull request by CupOfTea696**
Since that Laravel already comes with the `required_if` validation rule out of the box, I thought it would make sense to add a `required_unless` rule too. It would do the exact opposite, and require a field unless another field is set to value.

Additional changes in laravel/laravel needed if this is accepted.

### CODE REVIEW

File: src/Illuminate/Validation/Validator.php [hunk 1]
```diff
@@ -688,6 +688,29 @@ protected function validateRequiredIf($attribute, $value, $parameters)
     }
 
     /**
+     * Validate that an attribute exists when another attribute does not have a given value.
+     *
+     * @param  string  $attribute
+     * @param  mixed   $value
```
Comment (GrahamCampbell) [inline src/Illuminate/Validation/Validator.php]: for all new code, please don't align these

Comment (GrahamCampbell) [inline src/Illuminate/Validation/Validator.php]: just put 2 spaces either side of the type

Comment (CupOfTea696) [inline src/Illuminate/Validation/Validator.php]: Sorry, I just copied it over from the method above, hadn't realised it was aligned.

**Comment by CupOfTea696** [comment]
@taylorotwell shall I create a PR in laravel/laravel for the language file?

**Comment by taylorotwell** [comment]
I already added it. Thanks though!

On Sun, Nov 1, 2015 at 10:26 AM, Sven Wittevrongel <notifications@github.com

**Pull request merged**